Makefile

```diff
@@ -20,19 +30,34 @@ else
     $(error Couldn\'t find OpenCV)
 endif
 
-deepseg: $(TFLIBS)/libtensorflow-lite.a deepseg.cc loopback.cc transpose_conv_bias.cc
-	g++ $^ ${CFLAGS} ${LDFLAGS} -o $@
+# Output folder
+BIN=bin
+
+# Default target
+all: $(BIN) $(BIN)/deepseg
+
+clean:
+	-rm -rf $(BIN)
+
+$(BIN):
+	-mkdir -p $(BIN)
+
+# Primary binary - special deps
+$(BIN)/deepseg: $(TFLIBS)/libtensorflow-lite.a deepseg.cc loopback.cc transpose_conv_bias.cc
+	g++ $^ ${CFLAGS} ${TFCFLAGS} ${LDFLAGS} ${TFLDFLAGS} -o $@
 
 $(TFLIBS)/libtensorflow-lite.a: $(TFLITE)
 	cd $(TFLITE) && ./download_dependencies.sh && ./build_lib.sh
 
 $(TFLITE):
 	git submodule update --init --recursive
 
-all: deepseg
+# Single file test progs - OpenCV deps only
+$(BIN)/%: %.cc
```

Comment:
Why not make those rules also depend on `${BIN}`?

Reply:
Good point...